### DISCUSSION TITLE
Facebook login does nothing on iOS but works fine on Android

### DISCUSSION THREAD

**Discussion by Arandy** [capawesome-team/capacitor-firebase#416] [Q&A] [answered]
Hello, 

I'm trying to use Firebase Facebook signin on my ionic app. I've followed install guides accurately but whenever i call, in my code, this:

const result = await FirebaseAuthentication.signInWithFacebook();  

I see into the Xcode Log this message:

To Native ->  FirebaseAuthentication signInWithFacebook 10796655

and that's it.

Code execution stops there and does not continue towards the remaining code:

 async signInWithFacebook() {
    try{
      if(AppConfig.DEBUG) { console.log('FirebaseAuthentication'); }
      const result = await FirebaseAuthentication.signInWithFacebook();      
**<= stops here =>**
      if(AppConfig.DEBUG) { console.log('signInWithFacebook()',result); }
      this.user = result.user;
      if (result.credential.accessToken) {
        this.afterSignInWithFacebook(result.credential.accessToken);
      } else {
        if(AppConfig.DEBUG) { console.log("signInWithFB error", result); }
      }  
    }catch(e){
**<= never reaches this point =>**
      if(AppConfig.DEBUG) { console.log("signInWithFB error", e); }
  }
  };

No popup appears, just nothing.

I've checked my Facebook Developer Console and configuration, it seems all ok.

Do you have any clue on what could be the issue?

Thank you

**Comment by robingenz** [comment]
Thank you for your request!

1. Did you add `facebook.com` to the `providers` [configuration](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/authentication#configuration) array?
2. Did you add the `CapacitorFirebaseAuthentication/Facebook` pod to your `Podfile` (usually `ios/App/Podfile`)?

**Reply by Arandy** [reply, marked as answer]
1. yes
2. yes

It turns out....ionic was not creating capacitor.config.json. So it was not present on Xcode project. 
So I transformed my capacitor.config.ts into capacitor.config.json and copied into Xcode project.
Problem solved.

Now....why the hell Ionic did not create capacitor.config.json in the first place!!??!

I've gone mad for a whole day!!!!

Anyway, thank you form your answer @robingenz ;-)

**Reply by robingenz** [reply]
Glad you fixed it. But `capacitor.config.ts` should also work. I use the typescript version myself in a few projects.
